### PR TITLE
Accept Datatype Sensitive for git source URL

### PR DESCRIPTION
**Pull request by cocker-cc**
As the Source-String for the Git-Repo may contain sensitive Data, f.e.
an Access-Token, we now accept Datatype Sensitive.
Until the Command "exec" accepts Datatype Sensitive for "$cmd", we have
to unwrap here as a Workaround.

### CODE REVIEW

File: manifests/repo.pp
Comment (ekohl) [inline manifests/repo.pp:50]: I'm not sure this is really true? We still need to build a string and that needs to be unwrapped.

**Review by ekohl — APPROVED** [review]
I think the comment is incorrect, but generally the change looks correct.

File: manifests/repo.pp
Comment (cocker-cc) [reply manifests/repo.pp:50]: Yes, You are right, @ekohl.

Some Minutes ago, I learned, that `exec` indeed does accept `Sensitive` for `$command`.  
So the Way to go is to unwrap the Substring, and to wrap the Command-String again:  

```
  if $source and $source =~ Sensitive {
    $cmd = Sensitive.new("${bin} clone ${args_real} --recursive ${source.unwrap} ${target}")
  } elsif $source {
    $cmd = Sensitive.new("${bin} clone ${args_real} --recursive ${source} ${target}")
  } else {
    $cmd = "${bin} init ${args_real} ${target}"
  }
```

I will `commit --amend` this Change.

Comment (ekohl) [reply manifests/repo.pp:50]: I don' consider git URLs as sensitive and it can make debugging harder. Perhaps we should only make `$cmd` a `Sensitive` if the user passed in a `Sensitive.

Slight nit:

> ` if $source and $source =~ Sensitive {`

This is redundant. It's essentially the same as:
```puppet
if $source =~ NotUndef and $source =~ Sensitive {
```

Since `Sensitive is by definition `NotUndef`, `if $source =~ Sensitive {` should be sufficient.

Comment (cocker-cc) [reply manifests/repo.pp]: okay

**Review by ekohl — APPROVED** [review]
Thanks! I'll let tests complete before I merge, but I've been known to forget to check back. Feel free to ping if I do.

File: manifests/repo.pp
Comment (cocker-cc) [reply manifests/repo.pp]: I fixed to Your Suggestion.  

You stated: _"I don' consider git URLs as sensitive"_.   
With an API-Token, the URL is sensitive.  When the End-User (the one who writes Profile-Modules) has got such a sensitive URL, he should not bother with `unwrap`.  It should "just work".

Comment (ekohl) [reply manifests/repo.pp]: You're right: I should be clearer. In many cases the git URL isn't sensitive. We should continue making that easy to use. However, when the URL is sensitive, it should also be easy.

**Pull request merged**
